programs/kyupad-ido/src/instructions/update_config_project.rs

```diff
@@ -0,0 +1,97 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::TokenAccount;
+
+use crate::*;
+
+pub fn update_project_config(
+    ctx: Context<UpdateProjectConfig>,
+    update_config_project: UpdateProjectConfigArgs,
+) -> Result<()> {
+    let project = &mut ctx.accounts.project;
+
+    project.merkle_root = update_config_project.merkle_root;
+
+    match &update_config_project.total_ticket {
+        Some(total_ticket) => {
+            project.total_ticket = *total_ticket;
+        }
+        None => {}
+    }
+
+    Ok(())
+}
+
+pub fn update_destination(ctx: Context<UpdateDestination>, _project_id: String) -> Result<()> {
+    let project = &mut ctx.accounts.project;
+    let destination = &ctx.accounts.destination;
+
+    project.investment_destination = destination.key();
+    Ok(())
+}
+
+#[derive(Accounts)]
+#[instruction(update_config_project: UpdateProjectConfigArgs)]
+pub struct UpdateProjectConfig<'info> {
+    #[account(
+        mut,
+        constraint = creator.key() == admin_pda.admin_key
+    )]
+    pub creator: Signer<'info>,
+
+    #[account(
+        seeds=[b"admin", creator.key().as_ref()],  
+        bump,
+        owner = ID,
+    )]
+    /// CHECK:
+    pub admin_pda: Account<'info, Admin>,
+
+    #[account(
+        mut,
+        seeds = [ProjectConfig::PREFIX_SEED, update_config_project.project_id.as_bytes()],
+        bump,
+        owner = ID,
+    )]
+    pub project: Account<'info, ProjectConfig>,
+}
+
+#[derive(Accounts)]
+#[instruction(_project_id: String)]
+pub struct UpdateDestination<'info> {
+    #[account(
+        mut,
+        constraint = signer.key() == master_pda.master_key
```

Comment:
Is this spec? Can admin change the vault address?

Reply:
No sir